cmd/launcher/uninstall_darwin.go

```diff
@@ -0,0 +1,48 @@
+//go:build darwin
+// +build darwin
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+)
+
+func removeLauncher(ctx context.Context, logger log.Logger, identifier string) error {
+	if identifier == "" {
+		// Ensure identifier is non-empty and use the default if none provided
+		identifier = "kolide-k2"
+	}
+
+	launchDaemonPList := fmt.Sprintf("/Library/LaunchDaemons/com.%s.launcher.plist", identifier)
+	launchCtlPath := "/bin/launchctl"
+	launchCtlArgs := []string{"unload", launchDaemonPList}
+
+	if _, err := tablehelpers.Exec(ctx, logger, 30, []string{launchCtlPath}, launchCtlArgs); err != nil {
+		return err
+	}
+
+	// Now we can delete the plist which controls the launcher daemon
+	if err := os.Remove(launchDaemonPList); err != nil {
+		fmt.Printf("error removing file %s: %s\n", launchDaemonPList, err)
+	}
+
+	dirsToRemove := []string{
+		fmt.Sprintf("/usr/local/%s", identifier),
+		fmt.Sprintf("/etc/%s", identifier),
+		fmt.Sprintf("/var/%s", identifier),
+	}
```

Comment:
Would we want to clean up `/var/log/kolide-k2`? Probably not a big deal either way.

Reply:
We don't call it out in https://help.kolide.com/en/articles/2950178-removing-kolide-k2-launcher

...but maybe we should do it here? Unless there's value to those logs post uninstall.

Reply:
Ha. That, and `/etc/newsyslog.d/kolide-k2.conf` are somewhat undocumented and specific to macOS.

There's no value in keeping them, removing them isn't wrong.

Reply:
👍 Added these paths to the macOS uninstall